FigS2B: Mini ADT plots

allyhawkins opened the pull request:
Related to #8 

This PR adds in a script to create the mini versions of the ADT plots for the supplemental figure covering additional modalities. Similar to how we have mini and simplified versions of the plots in the main QC report, this creates similar looking figures but for the ADT plots. 

- I added some lines to the `sync-data-files.R` script to download the data files for `SCPCS000216` for creation of these plots.
- I used the same theme settings that we used to create the mini QC plots. 
- I removed extra text or lines that may be hard to read or confusing for these smaller versions. Specifically, I removed the N=X in the filtered plot showing the number of cells that are included. I also removed the dotted line, similar to how we did that in the simplified main QC reports. 
- I removed the legend for the UMAPs, again following suit with what we did before. 
- The code for the actual plots is pretty much copied from the `cite_qc.rmd` file in `scpca-nf` with some slight alterations. 

Again, I think it's going to be hard to fully decide on if we like the size until we put things in illustrator with the overview diagram, but I think this is a good start to at least get the look of the plots done.

Tagging @jaclyn-taroni for any aesthetic comments and then requesting @sjspielman for code review.

sjspielman commented on scripts/FigS2B_adt-plots.R:
```diff
@@ -0,0 +1,143 @@
+  ))
+
+# faceted plot to show cells belonging to each filter group
+filtered_plot <- ggplot(filtered_coldata_df, aes(x = detected, y = subsets_mito_percent, color = keep_column)) +
```
can you space this out onto multiple lines for readability?

sjspielman commented on scripts/FigS2B_adt-plots.R:
```diff
@@ -0,0 +1,143 @@
+
+# create data frame of UMAPs and expression
+umap_df <- scuttle::makePerCellDF(processed_sce) |>
+  tibble::rownames_to_column("barcode") |>
```
I feel like we might already actually have a column `barcodes` in the colData so you might be able to skip this, but definitely check if I'm right.

sjspielman commented on scripts/FigS2B_adt-plots.R:
```diff
@@ -0,0 +1,143 @@
+
+# Set up -----------------------------------------------------------------------
+
+# define file paths and read in sce objects
```
I might define `sample_id` and `library_id` first and then use below? But for a one-off, doesn't seem super necessary. Up to you.

sjspielman commented on scripts/FigS2B_adt-plots.R:
```diff
@@ -0,0 +1,143 @@
+
+# grab expression for top ADTs from counts
+var_adt_exp_df <- logcounts(altExp(processed_sce))[top_adts, ] |>
+  # as.matrix needs to come _first_
```
I too have learned this the hard way.

allyhawkins replied:
😂 I believe this is your original comment since I did copy some of this code

sjspielman replied:
Past me did a good job!!

sjspielman reviewed (commented):
Not a full review! Just some initial tiny thoughts from scrolling through! Actual review next week after aesthetic comments are in.

allyhawkins commented:
@sjspielman I responded to your initial comments. We do already have a `barcodes` column so I adjusted throughout to use that existing column and consistently use `barcodes` not `barcode`.

jaclyn-taroni commented on scripts/figure_setup/sync-data-files.R:
```diff
@@ -3,14 +3,22 @@
+adt_local_files <- here::here("s3_files", "SCPCS000216")
+fs::dir_create(adt_local_files)
+
+sync_call <- glue::glue("op run -- aws s3 cp '{adt_s3_files}' '{adt_local_files}' --exclude '*' --include '*.rds' --recursive")
```
I think you've broken with convention here:

```suggestion
sync_call <- glue::glue("aws s3 cp '{adt_s3_files}' '{adt_local_files}' --exclude '*' --include '*.rds' --recursive")
```

jaclyn-taroni commented on figures/pngs/FigS2B_adt-plots.png:
LGTM. Just flagging that I think we will need a very good legend for the filtering plot here!

sjspielman commented on scripts/FigS2B_adt-plots.R:
```diff
@@ -0,0 +1,144 @@
+theme_set(
+  theme_classic() +
+    theme(
+      strip.background = element_rect(fill = "transparent"),
```
The strip borders look a little thick in the final image, this seems good when I run on my end.

```suggestion
      strip.background = element_rect(fill = "transparent", linewidth = 0.5 ),
```

sjspielman commented on scripts/FigS2B_adt-plots.R:
```diff
@@ -0,0 +1,144 @@
+    )
+)
+
+set.seed(2024)
```
I don't think we need a seed in this script

sjspielman commented on scripts/FigS2B_adt-plots.R:
```diff
@@ -0,0 +1,144 @@
+
+combined_plot <- patchwork::wrap_plots(list(filtered_plot, adt_density_plot, adt_umap_plot))
+
+ggsave(output_plot_file, combined_plot)
```
When I run the script, the text doesn't fit - and it's because my exported figure size is different from yours. We'll need height & width arguments here to ensure everything fits, as well as associated reproducibility 

![FigS2B_adt-plots](https://github.com/AlexsLemonade/scpca-paper-figures/assets/4701111/ae52fb8f-c7e6-447a-a2fe-88fb99228eb7)

sjspielman reviewed (commented):
Mostly looks good, just a few small-ish comments!

While you're here, can you also add a line in the top comments in `generate-figure-tables.sh` to run `Rscript/scripts/sync-reference-files.R`? Thanks!

allyhawkins commented:
I updated the linewidth, set the figure size, and added a comment about running the set up script for reference files. This should be ready for another look @sjspielman.

sjspielman commented on figures/pngs/FigS2B_adt-plots.png:
Actually, I think maybe we might want to make the line around the density plot a bit thinner too?

sjspielman reviewed (approved):
LGTM! The only thing I might tweak is making the density plot outline a bit thinner, use your judgement! But, I don't think I need to see again unless you want me to look so going ahead and approving now.
